src/plugins/controls/public/control_group/actions/edit_control_flyout.tsx

```diff
@@ -40,9 +40,8 @@ export const EditControlFlyout = ({
 
   const [currentGrow, setCurrentGrow] = useState(panel.grow);
   const [currentWidth, setCurrentWidth] = useState(panel.width);
-  const [inputToReturn, setInputToReturn] = useState<Partial<DataControlInput>>({});
 
-  const onCancel = () => {
+  const onCancel = (inputToReturn: Partial<DataControlInput>) => {
```

Comment:
This is an old relic from when we were auto-applying the changes **behind the flyout** but, since we removed the behaviour, I think it's about time we cleaned this up :)

Basically, rather than calling the old `onTypeEditorChange` every time something in the flyout changes, I think it makes more sense to simply send the changes over a **single time** in the `onCancel` and `onSave` methods - that way, we don't have to update state twice every time something changes. Each part of the flyout can manage its own state and we can summarize these changes on save/cancel in order to apply them.